tests/test_pagination.py

```diff
@@ -0,0 +1,111 @@
+import tap_tester.connections as connections
+import tap_tester.menagerie   as menagerie
+import tap_tester.runner      as runner
+
+import os
+import logging
+from datetime import timedelta, date
+from datetime import datetime as dt
+from functools import reduce
+
+from base import TestSquareBase
+from test_client import TestClient
+
+class TestSquarePagination(TestSquareBase):
+    """Test that we are paginating for streams when exceeding the API record limit of a single query"""
+    API_LIMIT = 1000
+
+    def name(self):
+        return "tap_tester_square_pagination_test"
+
+    def testable_streams(self):
+        return self.expected_incremental_streams()
+
+    @classmethod
+    def setUpClass(cls):
+        print("\n\nTEST SETUP\n")
+        cls.client = TestClient()
+
+    def test_run(self):
+        """
+        Verify that for each stream you can get multiple pages of data
+        when no fields are selected and only the automatic fields are replicated.
+
+        PREREQUISITE
+        For EACH stream add enough data that you surpass the limit of a single
+        fetch of data.  For instance if you have a limit of 250 records ensure
+        that 251 (or more) records have been posted for that stream.
+        """
+        print("\n\nRUNNING {}\n\n".format(self.name()))
+
+        # Ensure tested streams have a record count which exceeds the API LIMIT
+        for stream in self.testable_streams():
+            existing_objects = self.client.get_all(stream, self.START_DATE)
+            if len(existing_objects) <= 1000:
```

Comment:
Should be data driven use var at top of test

Reply:
Can you clarify? We grab all records that we can, and check if we have enough (over 1000 records per stream).

Do you want new records every time?

Reply:
oh no just that you defined API_LIMIT at the top of the file and then didn't use it here